liblangutil/SemVerHandler.cpp

```diff
@@ -250,6 +252,18 @@ SemVerMatchExpression::MatchComponent SemVerMatchExpressionParser::parseMatchCom
 		else
 			break;
 	}
+
+	// Validate that the parsed version parts are either a single string literal or multiple bare tokens,
+	// i.e. "1.2.3" or 1.2.3 but not 1."2.3", "1".2.3 or 1"."2.3.
+	auto const partsEndPos = m_pos; // Points *after* the last version part
+	for (auto i = partsStartPos; i < partsEndPos; ++i)
```

Comment:
Do you even this loop and the positional tracking? I think you can have a counter/boolean in the above `while` loop, and check whether `currentToken() == Token::StringLiteral`, and then error out as soon as you have more than 1?